packages/headless/src/controllers/core/generated-answer/headless-core-generated-answer.ts

```diff
@@ -0,0 +1,327 @@
+import {generatedAnswerReducer as generatedAnswer} from '../../../features/generated-answer/generated-answer-slice';
+import {GeneratedAnswerState} from '../../../features/generated-answer/generated-answer-state';
+import {GeneratedResponseFormat} from '../../../features/generated-answer/generated-response-format';
+import {CoreEngine} from '../../../product-listing.index';
+import {
+  DebugSection,
+  GeneratedAnswerSection,
+  SearchSection,
+} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+import {
+  Controller,
+  buildController,
+} from '../../controller/headless-controller';
+
+export type {
+  GeneratedAnswerCitation,
+  GeneratedResponseFormat,
+  GeneratedAnswerState,
+};
+
+export interface GeneratedAnswer extends Controller {
+  /**
+   * The state of the GeneratedAnswer controller.
+   */
+  state: GeneratedAnswerState;
+  /**
+   * Re-executes the last query to generate an answer.
+   */
+  retry(): void;
+  /**
+   * Indicates that the generated answer met the user expectations.
+   */
+  like(): void;
+  /**
+   * Marks the generated answer as not relevant to the end user.
+   */
+  dislike(): void;
+  /**
+   * Re-executes the query to generate the answer in the specified format.
+   * @param responseFormat - The formatting options to apply to generated answers.
+   */
+  rephrase(responseFormat: GeneratedResponseFormat): void;
+  /**
+   * Opens the modal to provide feedback about why the generated answer was not relevant.
+   */
+  openFeedbackModal(): void;
+  /**
+   * Closes the modal to provide feedback about why the generated answer was not relevant.
+   */
+  closeFeedbackModal(): void;
+  /**
+   * Sends feedback about why the generated answer was not relevant.
+   * @param feedback - The feedback that the end user wishes to send.
+   */
+  sendFeedback(feedback: GeneratedAnswerFeedback): void;
+  /**
+   * Sends detailed feedback about why the generated answer was not relevant.
+   * @param details - Details on why the generated answer was not relevant.
+   */
+  sendDetailedFeedback(details: string): void;
+  /**
+   * Logs a custom event indicating a cited source link was clicked.
+   * @param id - The ID of the clicked citation.
+   */
+  logCitationClick(id: string): void;
+  /**
+   * Displays the generated answer.
+   */
+  show(): void;
+  /**
+   * Hides the generated answer.
+   */
+  hide(): void;
+  /**
+   * Logs a custom event indicating the generated answer was copied to the clipboard.
+   */
+  logCopyToClipboard(): void;
+  /**
+   * Logs a custom event indicating a cited source link was hovered.
+   * @param citationId - The ID of the clicked citation.
+   * @param citationHoverTimeMs - The number of milliseconds spent hovering over the citation.
+   */
+  logCitationHover(citationId: string, citationHoverTimeMs: number): void;
+}
+
+export interface GeneratedAnswerProps {
```

Comment:
```suggestion
export interface GeneratedAnswerCoreProps {
```
or 

```suggestion
export interface CoreGeneratedAnswerProps {
```
I would consider renaming the core props to avoid confusion

Reply:
Since those are currently re-exported as is in `headless-generated-answer` (and in turn, gets re-exported globally in index.ts), renaming those types would be a breaking change.